src/actors/objects/data_cache.rs

```diff
@@ -138,83 +138,85 @@ impl CacheItemRequest for FetchFileDataRequest {
-                    DownloadStatus::Completed => {
-                        log::trace!("Finished download of {}", cache_key);
-                        let decompress_result =
-                            decompress_object_file(&download_file, tempfile_in(download_dir)?);
-
-                        // Treat decompression errors as malformed files. It is more likely that
-                        // the error comes from a corrupt file than a local file system error.
-                        let mut decompressed = match decompress_result {
-                            Ok(decompressed) => decompressed,
-                            Err(_) => return Ok(CacheStatus::Malformed),
-                        };
-
-                        // Seek back to the start and parse this object so we can deal with it.
-                        // Since objects in Sentry (and potentially also other sources) might be
-                        // multi-arch files (e.g. FatMach), we parse as Archive and try to
-                        // extract the wanted file.
-                        decompressed.seek(SeekFrom::Start(0))?;
-                        let view = ByteView::map_file(decompressed)?;
-                        let archive = match Archive::parse(&view) {
-                            Ok(archive) => archive,
-                            Err(_) => {
-                                return Ok(CacheStatus::Malformed);
-                            }
-                        };
-                        let mut persist_file = fs::File::create(&path)?;
-                        if archive.is_multi() {
-                            let object_opt = archive
-                                .objects()
-                                .filter_map(Result::ok)
-                                .find(|object| object_id.match_object(object));
-
-                            let object = match object_opt {
-                                Some(object) => object,
-                                None => {
-                                    if archive.objects().any(|r| r.is_err()) {
-                                        return Ok(CacheStatus::Malformed);
-                                    } else {
-                                        return Ok(CacheStatus::Negative);
-                                    }
-                                }
-                            };
-
-                            io::copy(&mut object.data(), &mut persist_file)?;
-                        } else {
-                            // Attempt to parse the object to capture errors. The result can be
-                            // discarded as the object's data is the entire ByteView.
-                            if archive.object_by_index(0).is_err() {
-                                return Ok(CacheStatus::Malformed);
-                            }
 
-                            io::copy(&mut view.as_ref(), &mut persist_file)?;
-                        }
+        let future = async move {
```

Comment:
This was changed before I made `download()` require an `Arc`. Since the downloader (Arc) is moved into the async block, it's lifetime is contained within the future. The entire resulting block therefore did not expose the downloader's lifetime.

Since we now consume the `Arc`, there is no more reason to change this. However, the new block seems more readable with async await, which is why I kept it. In fact, some of the `tryf03pin!` calls could also be moved inside the block.